src/connectivity/mod.rs

```diff
@@ -659,6 +661,27 @@ pub fn digraph_all_pairs_all_simple_paths(
     ))
 }
 
+/// Return all the connected subgraphs (as a list of node indices) with exactly k nodes
+///
+///
+/// :param PyGraph graph: The graph to find all connected subgraphs in
+/// :param int k: The maximum number of nodes in a returned connected subgraph.
+///
+/// :returns: A list of connected subgraphs with k nodes, represented by their node indices
+///
+/// :raises ValueError: If ``k`` is larger than the number of nodes in ``graph``
+#[pyfunction]
+#[pyo3(text_signature = "(graph, k, /)")]
+pub fn connected_subgraphs(graph: &PyGraph, k: usize) -> PyResult<Vec<Vec<usize>>> {
```

Comment:
I wonder how much faster it is to return a Vector of `NodeIndices`, specially after #1096 and #1107 made them faster to use again from Python